pkg/app/provider/wopi/wopi.go

```diff
@@ -142,16 +142,29 @@ func (p *wopiProvider) GetAppURL(ctx context.Context, resource *provider.Resourc
 	}
 
 	q.Add("appname", p.conf.AppName)
-	q.Add("appurl", p.appURLs["edit"][ext])
 
-	if p.conf.AppIntURL != "" {
-		q.Add("appinturl", p.conf.AppIntURL)
+	var viewAppURL string
+	if viewAppURLs, ok := p.appURLs["view"]; ok {
+		if viewAppURL, ok = viewAppURLs[ext]; ok {
+			q.Add("appviewurl", viewAppURL)
+		}
 	}
-	if viewExts, ok := p.appURLs["view"]; ok {
-		if url, ok := viewExts[ext]; ok {
-			q.Add("appviewurl", url)
+	if editAppURLs, ok := p.appURLs["edit"]; !ok {
+		if editAppURL, ok := editAppURLs[ext]; ok {
+			q.Add("appurl", editAppURL)
 		}
+	} else {
+		// assuming that an view action is always available in the /hosting/discovery manifest
+		// eg. Collabora does support viewing jpgs but no editing
+		// eg. OnlyOffice does support viewing pdfs but no editing
+		// there is no known case of supporting edit only without view
```

Comment:
DrawIO I think is edit online but so far is not part of this model (yet).

Reply:
It could be a possibility to just pass 
`appurl` to the CS3 WOPI server, which is then the edit or open url depending on what action should be done (specified in the user request or supported by the app). Currently we also always set `appviewurl` which will not be used when the action is editing.

Reply:
As a matter of fact the full `CheckFileInfo` metadata does include both an edit and a view URL, and MS Office expects those. This is why both are expected from Reva regardless the action.

Reply:
Could we then omit the ones not specified? Looks to me like I should better fix this in the WOPI server? Didn't know that before

Reply:
As discussed offline, it should not be a problem to keep the whole discovery response as provided by WOPI.